File: project-{topic}-advanced/src/main/java/com/alx/webscraper/model/ScrapingTask.java
```java
package com.alx.webscraper.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

/**
 * Represents a single web scraping task defined by the user.
 * Contains configuration for what to scrape, where, and how often.
 */
@Entity
@Table(name = "scraping_tasks")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ScrapingTask {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name; // Name of the scraping task

    @Column(nullable = false)
    private String targetUrl; // The URL to scrape

    @ElementCollection(fetch = FetchType.EAGER) // Eagerly load data fields with the task
    @CollectionTable(name = "task_data_fields", joinColumns = @JoinColumn(name = "task_id"))
    private List<DataField> dataFields; // List of fields to extract, defined by CSS selectors

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ScrapingTaskStatus status; // Current status of the task (e.g., PENDING, RUNNING, COMPLETED, FAILED)

    @Column(nullable = true)
    private String cronExpression; // CRON expression for scheduling (e.g., "0 0 12 * * ?")

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @Column(nullable = true)
    private LocalDateTime lastRunAt;

    @Column(nullable = true)
    private String lastRunMessage; // Message from the last run, e.g., success or error

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private com.alx.webscraper.auth.model.User user; // User who owns this task

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
        if (this.status == null) {
            this.status = ScrapingTaskStatus.PENDING;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
```

File: project-{topic}-enhanced/src/entities/ScrapingTask.ts
```typescript
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  UpdateDateColumn,
  ManyToOne,
  JoinColumn,
  OneToMany,
} from "typeorm";
import { Project } from "./Project";
import { ScrapingResult } from "./ScrapingResult";

/**
 * @file ScrapingTask entity definition.
 *
 * This entity defines a single scraping task, including its target URL,
 * CSS/XPath selectors for data extraction, scheduling information,
 * and current status. Each task belongs to a project.
 */

// Define the structure for a selector configuration
export interface SelectorConfig {
  name: string;
  selector: string; // CSS selector or XPath
  type?: 'css' | 'xpath'; // Defaults to 'css'
  attribute?: string; // e.g., 'href' for anchor tags, 'src' for images
}

@Entity("scraping_tasks")
export class ScrapingTask {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @ManyToOne(() => Project, (project) => project.scrapingTasks)
  @JoinColumn({ name: "project_id" })
  project!: Project;

  @Column({ name: "target_url", type: "varchar", length: 255, nullable: false })
  targetUrl!: string;

  @Column({ type: "jsonb", nullable: false, default: [] })
  selectors!: SelectorConfig[];

  @Column({
    type: "varchar",
    length: 20,
    default: "pending",
    nullable: false,
  })
  status!: "pending" | "running" | "completed" | "failed" | "cancelled";

  @Column({ name: "schedule_interval", type: "varchar", length: 50, nullable: true })
  scheduleInterval!: string | null; // e.g., "daily", "weekly", "*/5 * * * *" (cron string)

  @Column({ name: "last_run_at", type: "timestamp", nullable: true })
  lastRunAt!: Date | null;

  @Column({ name: "next_run_at", type: "timestamp", nullable: true })
  nextRunAt!: Date | null;

  @Column({ type: "boolean", default: true, nullable: false })
  headless!: boolean; // Whether to run the browser in headless mode

  @OneToMany(() => ScrapingResult, (result) => result.task)
  scrapingResults!: ScrapingResult[];

  @CreateDateColumn({ name: "created_at" })
  createdAt!: Date;

  @UpdateDateColumn({ name: "updated_at" })
  updatedAt!: Date;
}
```